components/mdns/mdns.c

```diff
@@ -7541,9 +7556,20 @@ static void _mdns_browse_result_add_txt(mdns_browse_t *browse, const char *insta
                 !_str_null_or_empty(r->instance_name) && !strcasecmp(instance, r->instance_name) &&
                 !_str_null_or_empty(r->service_type) && !strcasecmp(service, r->service_type) &&
                 !_str_null_or_empty(r->proto) && !strcasecmp(proto, r->proto)) {
+            bool should_update = false;
             if (r->txt) {
+                // Check if txt changed
+                if (txt_count != r->txt_count) {
+                    should_update = true;
+                } else {
+                    for (size_t txt_index = 0; txt_index < txt_count; txt_index++) {
+                        if (!is_txt_item_exist(txt[txt_index], txt_value_len[txt_index], r->txt, r->txt_value_len, r->txt_count)) {
```

Comment:
are the txt items in the list supposed to be ordered the same way as in the record? If so, maybe we can give it an index to start the next `for()` loop in the function from there. That way we'd save some iterations for most common usecases, where nothing changes and we'd have to go over the list twice.

Reply:
TXT entries do not require a strict order, and entries in the old list and the new list may have the same content but in a different order. Therefore, it is necessary to traverse the entire list.

Reply:
What I meant was that perhaps a tiny optimization for the most common usecase where we receive the PTR of the same service (with the same TXT) record could be done, as in this case we'd get the TXT records in the same order.

I know that they could be arranged differently, but we can handle this common case faster if we start searching from the last outer index in the inner loop (also considering that the same service gets announced multiple times) ... but that would only be beneficial if we have like hunder(s) TXT record...

anyway, just a thought.